build.sh

```diff
@@ -88,12 +69,11 @@ for dir in ${dirs}; do
     docker_build_with_version Dockerfile
   fi
 
+  ok_to_tag=1
   if [[ -v TEST_MODE ]]; then
     VERSION=$dir IMAGE_NAME=${IMAGE_NAME} test/run
-
-    if [[ $? -eq 0 ]] && [[ "${TAG_ON_SUCCESS}" == "true" ]]; then
-      echo "-> Re-tagging ${IMAGE_NAME} image to ${IMAGE_NAME%"-candidate"}"
-      docker tag $IMAGE_NAME ${IMAGE_NAME%"-candidate"}
+    if [[ $? -ne 0 ]] || [[ "${TAG_ON_SUCCESS}" != "true" ]]; then
```

Comment:
I think '[[ $? -eq 0 ]]' used to be no-op before, because if 'test/run' failed, the 'set -e' would already exit the shell anyways right before the condition....   And now the reversed version '[[ $? -ne 0 ]]' might never succeed too.